src/YarpUtilities/src/RosPublisher.cpp

```diff
@@ -0,0 +1,482 @@
+    return true;
+}
+
+bool RosPublisher::configureTransformServer(const std::string& transformServerPort)
+{
+    if (m_pimpl->transformBroadcaster.isValid())
+    {
+        m_pimpl->transformBroadcaster.close();
+        m_pimpl->transformInterface = nullptr;
+    }
+    
+    yarp::os::Property tfBroadcasterSettings{{"device", yarp::os::Value("transformClient")},
+                                             {"remote", yarp::os::Value(transformServerPort)},
+                                             {"local", yarp::os::Value(m_pimpl->nodeName + "/transformClient")}};
+
+    std::string_view printPrefix = "[RosPublisher::openTransformBroadcaster] ";
+    if (!m_pimpl->transformBroadcaster.open(tfBroadcasterSettings))
+    {
+        std::cerr << printPrefix << "Unable to open transform broadcaster. Ensure transform server is already running." << std::endl;   
+        return false;
+    }
+
+    if (!m_pimpl->transformBroadcaster.view(m_pimpl->transformInterface))
+    {
+        std::cerr << printPrefix << "Unable to access transform interface." << std::endl; 
+        return false;
+    }
+    
+    return true;
+}
+
+bool RosPublisher::publishTransform(const std::string& target, 
+                                    const std::string& source, 
+                                    const BipedalLocomotion::GenericContainer::Vector<const double>::Ref transformAsVector16d)
+{
+    std::string_view printPrefix = "[RosPublisher::publishTransform] ";
+    
+    if (!m_pimpl->initialized)
+    {
+        std::cerr << printPrefix << "Please call initialize before publishing." << std::endl;
+        return false;
+    }
+    
+    if (!m_pimpl->publishTF || m_pimpl->transformInterface == nullptr)
+    {
+        std::cerr << printPrefix << "Transform broadcaster was not configured. Unable to publish transforms." << std::endl;
+        return false;
+    }
+    
+    if (transformAsVector16d.size() != 16)
+    {
+        std::cerr << printPrefix << "Malformed Transform data. Expecting a 16d vector." << std::endl;
+        return false;
+    }
+    
+    yarp::sig::Matrix pose(4, 4);
+    for (size_t i = 0; i < 4; i++)
+    {
+        for (size_t j = 0; j < 4; j++)
+        {
+            pose(i, j) = transformAsVector16d( (i*4) + j);
+        }
+    }
+    
+    if (!m_pimpl->transformInterface->setTransform(target, source, pose))
+    {
+        std::cerr << printPrefix << "Could not publish transform for source: " 
+                  << source << " and target: " << target << " frames." << std::endl;
+        return false;
+    }
+
+    return true;
+}
+
+bool RosPublisher::publishJointStates(const GenericContainer::Vector<const std::string>::Ref jointList,
+                                      const GenericContainer::Vector<const double>::Ref jointPositions)
+{        
+    std::vector<double> jointVelocities(jointList.size(), 0.0);
+    std::vector<double> jointEfforts(jointList.size(), 0.0);
```

Comment:
Actually, here you don't need two separate vectors. In addition, consider having a dummy buffer of the correct size as class method in order to avoid unnecessary memory allocation.

Reply:
Addressed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/156/commits/d57a286f4780f87c335ff0b3b91ca01443ffa069